FHEM/00_SIGNALduino.pm

```diff
@@ -4320,42 +4392,35 @@ sub SIGNALduino_FW_getProtocolList
 			$msgtype = "MU";
 		}
 		
-		my $cmd;
-		my $newWlIDs;
-		if ( (grep { $_ eq $id } @{$hash->{msIdList}}) ||  (grep { $_ eq $id } @{$hash->{muIdList}}) || (grep { $_ eq $id } @{$hash->{mcIdList}}) ) 
-		{
-			$cmd = FW_makeImage("on","disable","icon");
-			if ($wl_attr eq ".*")
-			{
-				$newWlIDs = join(",",@{$hash->{msIdList}},@{$hash->{muIdList}},@{$hash->{mcIdList}}) 
-			} else  {
-				$newWlIDs=$wl_attr;
-			}
-
-			$newWlIDs =~ s/(?:^$id,?|,$id,|$id$)/,/;
+		my $checked="";
 		
-		} else {
-			$cmd = FW_makeImage("off","enable","icon");
-			$newWlIDs = defined($wl_attr) ? "$wl_attr,$id" : $id;
+		if (exists($activeIdHash{$id}))
+		{
+			$checked="checked";
 		}
 		
+		if ($devFlag == 0 && exists($ProtocolListSIGNALduino{$id}{developId}) && $ProtocolListSIGNALduino{$id}{developId} eq "p") {
+			$chkbox="<div> </div>";
+		}
+		else {
+			$chkbox=sprintf("<INPUT type=\"checkbox\" name=\"%s\" %s/>",$id,$checked);
```

Comment:
In dem Feld Name sollten keine Namen mit ".,!?" enthalten.

Reply:
Der name ist hier die id. Die id kann nicht  ",!?" enthalten.
Der Punkt ist aber in der id, was kann passieren, wenn ein Punkt im Namen enthalten ist?